reporter/src/main/kotlin/reporters/opossum/OpossumReporter.kt

```diff
@@ -501,15 +502,15 @@ class OpossumReporter : Reporter {
     override val type = "Opossum"
 
     private fun writeReport(outputFile: File, opossumInput: OpossumInput) {
-        FileOutputStream(outputFile, /* append = */ false).use { outputStream ->
-            val gzipParameters = GzipParameters().apply {
-                compressionLevel = Deflater.BEST_COMPRESSION
-            }
-
-            GzipCompressorOutputStream(outputStream, gzipParameters).bufferedWriter().use { gzipWriter ->
-                JsonMapper()
-                    .setSerializationInclusion(Include.NON_NULL)
-                    .writeValue(gzipWriter, opossumInput.toJson())
+        ZipOutputStream(BufferedOutputStream(FileOutputStream(outputFile))).use { zipOutputStream ->
+            val jsonMapper = ObjectMapper()
+            val jsonBytes = jsonMapper.writeValueAsBytes(opossumInput.toJson())
+            val byteArrayInputStream = ByteArrayInputStream(jsonBytes)
+            BufferedInputStream(byteArrayInputStream).use { bufferedInputStream ->
+                val inputJsonEntry = ZipEntry("input.json")
+                zipOutputStream.putNextEntry(inputJsonEntry)
+                bufferedInputStream.copyTo(zipOutputStream, 4096)
+                zipOutputStream.closeEntry()
```

Comment:
If you're fine with creating a temporary file, you could probably do instead (untested):

        val jsonFile = createOrtTempFile()
        JsonMapper().setSerializationInclusion(Include.NON_NULL).writeValue(jsonFile, opossumInput.toJson())
        jsonFile.packZip(outputFile)
        jsonFile.delete()

where `packZip()` comes from ORT's own `ArchiveUtils`.

Reply:
Nice and elegant suggestion. Works like a charm. Thanks.

Reply:
> Works like a charm.

Except for the `prefix` bit, see my comment below.